footbot/research/utils/simulator.py

```diff
@@ -48,54 +48,53 @@ def get_all_results_df(season, client):
     return df
 
 
-def aggregate_predictions(predictions_df, start_event, end_event):
+def aggregate_predictions(predictions_df, start_event, end_event, weight):
     """
-    Averages point predictions for multiple gameweeks.
+    Calculates weighted sum of point predictions for multiple gameweeks.
     :param predictions_df: Dataframe of points predictions by player, gameweek
     :param start_event: First gameweek event
     :param end_event: Last gameweek event
+    :param weight: weight for exponential decay of points
     :return: Dataframe of points predictions by player
     """
 
-    num_events = end_event - start_event + 1
-
     predictions_df = predictions_df.copy()
     predictions_df = predictions_df.loc[
         predictions_df["event"].between(start_event, end_event), :
     ]
 
+    predictions_df['weight'] = weight ** (predictions_df['event'] - start_event)
```

Comment:
it might be good to put

```
assert predictions['weight'].min() >= start_event
```

just so it crashes instead of giving you weird results

Reply:
What are you trying to catch here? That the first event in the dataframe is after the start event? If so, that's taken care of by line 63.

Reply:
ah i see yes. very good